scripts/check_commit_metadata.sh

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env bash
+
+# Usage: source ./check_commit_tags.sh <revision range>
+# Usage: ./check_commit_tags.sh <revision range>
+#
+# Example: ./check_commit_tags.sh v0.13.1..971e3678
+#
+# When sourced, this script will populate the COMMIT_METADATA_* variables
```

Comment:
_Applicable only to squash & merge_ - Do you think that we may need another script executed in CI to check if the title of PR is consistent?

Reply:
Hmm, what do you mean by _Applicable only to squash & merge_? I guess we could make it clearer this script is only intended for the commit log on the `main` branch.

Yes, we'll add an action/lint for PR title 👍🏻

Reply:
> Yes, we'll add an action/lint for PR title 👍🏻 (follow up PR)

👍 

> Applicable only to squash & merge?

I thought that `Create a merge commit` option is also enabled for this repo, but it seems to be off. If it's on, it might be more challenging to validate also commits than only the PR title.

Reply:
Ah right, yeah we'll only validate PR titles, I made the assumption when writing that we'll only squash merge.

Obviously owners/admins could bypass this by pushing straight to `main`. 🙄